### ISSUE TITLE
`transform` with `nullish, nullable` incorrectly infers

### ISSUE THREAD

**Issue by husseyexplores** [fabian-hiller/valibot#195]
`transform` does not infer type correctly when combined with `nullish`.

```ts
const Thing = transform(nullish(object({ id: string() })), x => {
    return 2 as number
})
type Thing = Output<typeof Thing>
```
The inferred type should be `number`. But we get `{ id: string }`.
Removing `nullish` results in correct types.

This is also the case with `nullable` as well. 

Live examples: https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbwFBzhARgKwKYGMYA0KcAdgK4jrZRGoDOMUwJA5rXIwIYl0Bm0IduQA2w4HQAW7GAE8w2OAHkyMMCvacoUTjKIBfOLygQQcAOQA3TmPQQYZpElCRYiOAC84BoyfPuIACYOSLgQPPAAKhLMLHAAvBzaPPxQIAAUImKSaRg4+GkIcMABAFxwDEysaQCUXtXVBHAAHvEAfIjEqFDYMGRQJHAATHCcdKQUVFBIetVIsvJwUTHxSipqMAA889gQvIvRrK2dcAD0JwB6APyOZ3AAjAB0cADuwDAScAAGmeISn0i3bBNeT4bClL4VGIAbQAup82nA0gAxACCAEkADKzULhOAorQ6ADKjBiaJIoS0eHgCS4yQEGTIol+aUhVXqjRacXayFQqGAezSAEImrVur1+nBYSMxvjtDINqyWEdeXAxX0Bk0HnQwGIYGkzI0zLMZnM5ApZUSSawyRTuvgVspVCotmbdniCTJiZUWDboHaYMqVbcro4ASchk9Xu8ICovj9JP9bgBhRQAJVTAFEkxE4ABaOBAkEwMFlT6K2HwrmIgAKKMJhOxYQY7rlXpiSb9VJWtL49MVNQ5CJ5vP5iOFop66slMOlLZ0CqtSuOaolmu1uv1huNs22c89i47lPtCUd6xd8jdFv33sP-sDqGD1xu4YAzJG3h9voysn8wwXgVSJZxhM1CVu0yLoliIRNvAKIkHYEjUL6R7Uok3C9qkDJMtkuRUgURTgv21QzA0zRDsuk4SsMozjJQ1DTDuZp4vB7xIeSnbHqsTqbNsl4sYhUDIXexynOGIbPnAAAs77RrGX7Yb+gIAaC4JloucAAD7lIuFYImktb1o2uIosIdAQLeXY0kkGHpERg5VsOfICuOqqUQMUo0VeC7evernihqDGmgsJlmRZnGns6vF7CF5kcQGnSPqGty5ilqVpelGWZVlKWODizYAFqBBmTScOAwgKAk7haouNQPPGEi1T2KTpJy3LHKOQoin5U4eWM5YwhR-nNFqOpvJu5jbkFCiFQExWlTqFUeA8zC8NQ547HsM1zWV2BHEAA

**Comment by fabian-hiller** [comment]
Thank you! I will fix that with the next version.

**Comment by fabian-hiller** [comment]
Will be fixed with the next version! Thanks again!

**Comment by fabian-hiller** [comment]
Fixed in v0.19.0
